tests/transaction_tests.rs

```diff
@@ -0,0 +1,401 @@
+                assert_eq!(filler.expect.len(), 0);
+                panic!("Decoding of this RLP data should fail");
+            }
+
+            data
+        }
+        Err(e) => {
+            panic!("Decoding failed correctly with {:?}", e);
+            return;
+        }
+    };
+    // A valid decoded transaction has exactly 9 elements.
+    assert_eq!(data.len(), 9);
+
+    let decoded_tx = Transaction {
+        nonce: (&*data[0]).into(),
+        gas_price: (&*data[1]).into(),
+        gas_limit: (&*data[2]).into(),
+        to: (&*data[3]).into(),
+        value: (&*data[4]).into(),
+        data: (&*data[5]).into(),
+        signature: Some(Signature::new(
+            (&*data[6]).into(),
+            (&*data[7]).into(),
+            (&*data[8]).into(),
+        )),
+    };
+
+    // We skipped all fillers without transaction data, so now this unwrap is safe.
+    let raw_params = filler.transaction.as_ref().unwrap();
+    // Create a tx based on filler params
+    let tx = Transaction {
+        nonce: raw_params.nonce.parse().unwrap_or(BigEndianInt::zero()),
+        gas_price: raw_params.gas_price.parse().unwrap_or(BigEndianInt::zero()),
+        gas_limit: raw_params
+            .gas_limit
+            .parse()
+            .expect("Unable to parse gas_limit"),
+        to: raw_params.to.parse().expect("Unable to parse address"),
+        value: raw_params.value.parse().unwrap_or(BigEndianInt::zero()),
+        data: hex_str_to_bytes(&raw_params.data).expect("Unable to parse data"),
+        signature: Some(Signature::new(
+            raw_params.v.parse().expect("Unable to parse v"),
+            raw_params.r.parse().expect("Unable to parse r"),
+            raw_params.s.parse().expect("Unable to parse s"),
+        )),
+    };
+
+    // Compare decoded transaction based on RLP and a transaction based on TX.
+    // No need to go through validation for `decoded_tx` since we can rely on the equality here,
+    // and assume if tx is valid, then decoded_tx is valid as well.
+    assert_eq!(decoded_tx, tx);
+
+    // Encoding of our transaction
+    let our_rlp = to_bytes(&tx).unwrap();
+    // All rlp's Fixtures
+    assert!(fixtures.rlp.starts_with("0x"));
+
+    assert!(tx.is_valid(), "{:?} {:?} {:?}", tx, raw_params, filler);
+    assert!(
+        tx.signature.as_ref().unwrap().is_valid(),
+        "{:?} {:?} {:?}",
+        tx.signature.as_ref().unwrap(),
+        raw_params,
+        filler
+    );
+    // Comparing our encoding with the "ground truth" in the fixture
+    assert_eq!(
+        bytes_to_hex_str(&our_rlp),
+        &fixtures.rlp[2..],
+        "{:?} != {:?} (filler {:?})",
+        &tx,
+        &raw_params,
+        &filler
+    );
+
+    // We have verified that case already so unwrapping an expect data is safe.
+    let expect = expect.expect("Expect should be available at this point");
+
+    // TODO: Change v to u64 so it would validate overflow when decoding/creating (v <= 2**64-1 so it can't overflow)
```

Comment:
this shouldn't take more than a second right?

Reply:
I gave it a shot and it wasn't that straightforward to just replace it. I had many errors with RLP encoding, which suggest it might be something with how serde-rlp handles u64.

Reply:
I opened an issue for that problem: https://github.com/althea-mesh/serde-rlp/issues/10. Once its solved we can move forward with changing this to u64.